Model/Config/Backend/Installments.php

```diff
@@ -112,8 +112,8 @@ public function beforeSave()
     protected function _afterLoad()
     {
         $value = $this->getValue();
-        $value = $this->serializer->unserialize($value);
-        if (is_array($value)) {
+        if (!empty($value) && is_array($value)) {
```

Comment:
the is_array check should happen after the unserialize because the $value which is serialized will always be a string.

Reply:
yes this now make sense but when $value returns null there’s no empty check for it.

Reply:
I agree with the empty check. This is what I meant:

```
if (!empty($value)) {
    $value = $this->serializer->unserialize($value);
    if (is_array($value)) {
        $value = $this->encodeArrayFieldValue($value);
        $this->setValue($value);
    }
}
```

Reply:
Correct - That should do!
However that will increase your method’s, eventually class’ cognitive complexities over all.

May be - we can return if $value is empty in the first place and focus on executing other statements?

Reply:
I'm okay with that.